lib/place_mark.rb

```diff
@@ -23,6 +21,14 @@ def execute(player:, x:, y:)
   private
 
   def check_dup?(x, y)
-    @board[x][y] == :O || @board[x][y] == :X
+    coordinate_has_O?(x,y) || coordinate_has_X?(x,y)
```

Comment:
Extracting these two conditionals as methods definitely help with readability here.

Reply:
With this check_dup? method, are you checking if a mark has already been placed before placing a new one?
If so, rather than checking if there is an :O or an :X already there. 
Could you maybe just check for a '-'? If there isn't a '-', we can imply that there has to be a :O or :X.

e.g:
def check_dup?(x, y)
  @board[x][y] != '-'
  # check if the co-ordinates do not contain a '-'
end

Reply:
Thanks we'll try that